### PR TITLE
Adding NL BR KVK 2.02 validation

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Reason for change

Added NL BR KVK 2.02 validation: FR-KVK-2.02: The attribute 'xml:lang' MUST contain the same value within an XBRL
instance document 

#### Steps to Test

Run validation against  tests/resources/validation/NL/fr_kvk/2.02/pass/valid.xml  and verify NO KVK 2.02 errors are returned. 

Run validation against  tests/resources/validation/NL/fr_kvk/2.02/fail/invalid.xml  and verify ONE KVK 2.02 error is returned.
**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/NL/rules/fr_kvk.py [hunk 1]
```diff
@@ -120,6 +120,36 @@ def rule_fr_kvk_2_03(
...
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-KVK-2.02: The attribute 'xml:lang' MUST contain the same value within an XBRL instance document
+    """
+    modelXbrl = val.modelXbrl
+    doc_xml_lang = []
+    for fact in modelXbrl.facts:
+        if fact.xmlLang and fact.xmlLang not in doc_xml_lang:
+            doc_xml_lang.append(fact.xmlLang)
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/NL/rules/fr_kvk.py:143]: This should presumably also check footnotes and the lang attribute of the containing xbrl element. Also, Arelle is written with camel case variables, not the typical Python snake case.

File: tests/resources/conformance_suites/nl_nt16/fr_kvk/2-02-multiple-lang-fact.xbrl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        xmlns:jenv-bw2-i="http://www.nltaxonomie.nl/nt16/jenv/20211208/dictionary/jenv-bw2-data"
+        xmlns:kvk-i="http://www.nltaxonomie.nl/nt16/kvk/20211208/dictionary/kvk-data"
+        xmlns:link="http://www.xbrl.org/2003/linkbase"
+        xmlns:nl-cd="http://www.nltaxonomie.nl/nt16/sbr/20210301/dictionary/nl-common-data"
+        xmlns:xbrli="http://www.xbrl.org/2003/instance"
+        xmlns:xlink="http://www.w3.org/1999/xlink">
+    <link:schemaRef
+            xlink:href="http://www.nltaxonomie.nl/nt16/kvk/20211208/entrypoints/kvk-rpt-jaarverantwoording-2021-nlgaap-klein-publicatiestukken.xsd"
+            xlink:type="simple"/>
+    <context id="ctx-1">
```
Comment (aaroncameron-wk) [inline tests/resources/conformance_suites/nl_nt16/fr_kvk/2-02-multiple-lang-fact.xbrl:15]: This context is incomplete, which is preventing your validation from firing properly.

File: arelle/plugin/validate/NL/rules/fr_kvk.py [hunk 3]
```diff
@@ -82,6 +86,56 @@ def rule_fr_kvk_2_01(
...
+            docXmlLang.append(fact.xmlLang)
+    assert modelXbrl.modelDocument is not None
+    if modelXbrl.modelDocument.type in (ModelDocument.Type.INLINEXBRL, ModelDocument.Type.INLINEXBRLDOCUMENTSET, ModelDocument.Type.UnknownXML):
+        for ixdsHtmlRootElt in modelXbrl.ixdsHtmlElements:
+            for elt in etree_iter(cast(_Element, ixdsHtmlRootElt)):
+                if isinstance(elt, ModelInlineFootnote) or (isinstance(elt, ModelResource) and elt.qname == XbrlConst.qnLinkFootnote):
+                    if elt.xmlLang and elt.xmlLang not in docXmlLang:
+                        docXmlLang.append(elt.xmlLang)
+    elif modelXbrl.modelDocument.type == ModelDocument.Type.INSTANCE:
+        for elt in modelXbrl.modelDocument.xmlRootElement.iter():
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/NL/rules/fr_kvk.py:125]: This is skipping over the root element itself, which is supposed to define what the language is for the entire document. See FR-KVK-2.01. I think a preferred approach would be to retrieve the root language first, and `yield` a validation error for every instance of a mismatching language within the doc, that way the error can include document and line number for each occurrence.

File: arelle/plugin/validate/NL/rules/fr_kvk.py [hunk 4]
```diff
@@ -82,6 +86,56 @@ def rule_fr_kvk_2_01(
...
+    """
+    FR-KVK-2.02: The attribute 'xml:lang' MUST contain the same value within an XBRL instance document
+    """
+    modelXbrl = val.modelXbrl
+    docXmlLang = []
+    for fact in modelXbrl.facts:
+        if fact.xmlLang and fact.xmlLang not in docXmlLang:
+            docXmlLang.append(fact.xmlLang)
+    assert modelXbrl.modelDocument is not None
+    if modelXbrl.modelDocument.type in (ModelDocument.Type.INLINEXBRL, ModelDocument.Type.INLINEXBRLDOCUMENTSET, ModelDocument.Type.UnknownXML):
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/NL/rules/fr_kvk.py:118]: My reasoning could be off here, but I don't think we need to support inline XBRL here. FR-KVK-1.01 explicitly requires an `.xbrl` file extension and nothing in the KVK rules mentions inline XBRL that I've seen.

Comment (derekgengenbacher-wf) [reply arelle/plugin/validate/NL/rules/fr_kvk.py:118]: Aarons assessment is correct. There is no ixbrl with this type of filing at the moment.

**Review by aaroncameron-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**